test/functional/ui/screen_basic_spec.lua

```diff
@@ -75,9 +76,19 @@ describe('Screen', function()
       local function check()
         eq(true, screen.suspended)
       end
+
+      -- when we use eval in the test, nvim instance is already foregrounded
+      -- after being suspended, therefore both events are triggered here.
+      command('autocmd VimSuspendPre  * :let g:foo=42')
+      command('autocmd VimResumed  * :let g:bar=24')
+
       command('suspend')
+      eq(42, eval('g:foo'))
+      eq(24, eval('g:bar'))
+
       screen:expect(check)
       screen.suspended = false
+
       feed('<c-z>')
```

Comment:
also check after `<c-z>`. (should do something like:

    command('autocmd VimSuspendPre  * :let g:suspend=1+get(g:, "suspend", 0)')

so that `g:suspend` is incremented each time the event fires.

Reply:
that's a nice idea! never thought of it that way. will make the changes mentioned here and in the previous comment.